gc.c

```diff
@@ -3287,38 +3288,40 @@ heap_page_for_ptr(rb_objspace_t *objspace, uintptr_t ptr)
         return NULL;
     }
 }
-#endif
 
 PUREFUNC(static inline int is_pointer_to_heap(rb_objspace_t *objspace, void *ptr);)
 static inline int
 is_pointer_to_heap(rb_objspace_t *objspace, void *ptr)
 {
-#ifdef USE_THIRD_PARTY_HEAP
-    if (ptr == NULL) {
-     	RUBY_DEBUG_LOG("      %18p: It is NULL. Nope.", ptr);
-        return false;
-    }
+#if USE_MMTK
+    if (mmtk_enable) {
+        if (ptr == NULL) {
+            RUBY_DEBUG_LOG("      %18p: It is NULL. Nope.", ptr);
+            return false;
+        }
 
-    // The granularity of MMTk's alloc_bit bitmap is one bit per word.
-    // We must check for alignment before asking MMTk.
-    if ((uintptr_t)ptr % sizeof(void*) != 0) {
-     	RUBY_DEBUG_LOG("      %18p: Not properly aligned. Nope.", ptr);
-        return false;
-    }
+        // The granularity of MMTk's alloc_bit bitmap is one bit per word.
+        // We must check for alignment before asking MMTk.
+        if ((uintptr_t)ptr % sizeof(void*) != 0) {
+            RUBY_DEBUG_LOG("      %18p: Not properly aligned. Nope.", ptr);
+            return false;
+        }
 
-    // Now let MMTk decide.
-    bool result = mmtk_is_mmtk_object(ptr);
+        // Now let MMTk decide.
+        bool result = mmtk_is_mmtk_object(ptr);
 
-    if (USE_RUBY_DEBUG_LOG) {
-	if (result) {
-	    RUBY_DEBUG_LOG("***** %18p: YEAH! It looks like an object reference! *****", ptr);
-	} else {
-	    RUBY_DEBUG_LOG("      %18p: MMTk says nope.", ptr);
-	}
+        if (USE_RUBY_DEBUG_LOG) {
+        if (result) {
+            RUBY_DEBUG_LOG("***** %18p: YEAH! It looks like an object reference! *****", ptr);
+        } else {
+            RUBY_DEBUG_LOG("      %18p: MMTk says nope.", ptr);
+        }
+        }
```

Comment:
These mis-indented lines look like a result from expanding tabstops to spaces last time.